streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -92,69 +95,80 @@ public void validatePartitionExists(final TopicPartition topicPartition, final S
 
     @Override
     public void register(final StateRestorer restorer) {
-        if (restorer.offsetLimit() > 0) {
-            stateRestorers.put(restorer.partition(), restorer);
+        stateRestorers.put(restorer.partition(), restorer);
+    }
+
+
+    public Collection<TopicPartition> restore() {
+        if (!initialized) {
+            initialize();
+        }
+
+        if (needsRestoring.isEmpty()) {
```

Comment:
I see.. it only happens if no tasks has any states so no restoration is needed, right? If that's the case can we move it inside the condition of line 103 to make it more obvious?